docs/dev/README.md

```diff
@@ -1,4 +1,90 @@
-# Permanently Moved
+# Developing Cluster Version Operator
 
-Other than run-cvo-locally.md, all other documents have moved to [openshift/enhancements](https://github.com/openshift/enhancements/tree/master/dev-guide/cluster-version-operator).
+> :warning: This document only covers the basic development, building and testing tasks. For information about
+> concepts used and implemented in CVO, please see the
+> [dev-guide in the openshift/enhancements repository](https://github.com/openshift/enhancements/tree/master/dev-guide/cluster-version-operator).
 
+## Building CVO
+
+### Building the CVO binary
+
+```console
+$ make build
+hack/build-go.sh
+Using version from git...
+Building github.com/openshift/cluster-version-operator (v1.0.0-942-g59e0826f-dirty)
+```
+
+### Building the CVO image
+
+The [`hack/build-image.sh`](../../hack/build-image.sh) script builds a local CVO image:
+
+```console
+$ ./hack/build-image.sh
+INFO: Using version from git...
+...
+[2/2] COMMIT cluster-version-operator:v1.0.0-942-g59e0826f-dirty
+--> 98ff149964f
+Successfully tagged localhost/cluster-version-operator:v1.0.0-942-g59e0826f-dirty
+```
+
+### Publishing a development CVO image
+
+The [`hack/push-image.sh`](../../hack/push-image.sh) script publishes
+a [locally-built CVO image](#building-the-cvo-image) to a remote repository:
+
+```console
+$ REPO=<your personal repo, such as quay.io/login> ./hack/push-image.sh
+```
+
+This pushes `${VERSION}` and `latest` tags to `${REPO}/origin-cluster-version-operator`, where `${VERSION}` encodes the
+Git commit used to build the image.
+
+### Building and publishing a release payload image with development CVO
+
+After publishing a [development CVO image](#publishing-a-development-cvo-image), you can build a release payload image
+that will contain all manifests from an existing release payload and the development CVO binary:
+
+```console
+$ REPO=<your personal repo, such as quay.io/login> ./hack/push-image.sh
```

Comment:
down in _Replace cluster's CVO with development CVO release payload image_, you list publishing this image as a prereq, and do not repeat the steps to publish.  Here, you list publishing this image as a prereq and are repeating the steps to publish.  I'm fine with both inline copies of the command, or expecting folks to go read the CVO-image-publishing section to get the command, but it seems like both CVO-image-dependent sections should use the same approach.

Reply:
Thanks, addressed in 7971a6a96def102c258193b3a768ab075a33d62f by removing the repeated publishing steps. I also noticed the crosslink in _Replace cluster's CVO with development CVO release payload image_ was leading to a wrong section, so fixed that too.